commands-yml/validator.js

```diff
@@ -0,0 +1,51 @@
+import validate from 'validate.js';
+
+validate.validators.array = function (value, options, key, attributes) {
+  if (attributes[key] && !validate.isArray(attributes[key])) {
+    return `must be an array`;
+  }
+};
+
+validate.validators.hasAttributes = function (value, options) {
+  if (!value) {
+    return;
+  }
+
+  if (typeof value === 'object' && !Array.isArray(value)) {
+    value = [value];
+  }
+
+  for (let item of value) {
+    for (let option of options) {
+      if (!item[option]) {
+        return `must have attributes: ${options}`;
+      }
+    }
+  }
+};
+
+export default {
+  'name': { 'presence': true },
```

Comment:
according to our style, we don't want spaces around braces here, and we also don't want string quotes unless necessary:

```js
name: {presence: true},
...
'client_docs.ruby': {presence: true, url: true},
...
```